### ISSUE TITLE
Failed to build statgen/savvy

### ISSUE THREAD

**Issue by Welz2** [statgen/savvy#31]
Hi this error keeps popping up and I'm unsure why.
I am using Mac M1- macOS Ventura
This is trying to download Saige-gene plus overall

configure: error: in `/Users/wellaglover/Documents/Codes RP/savvy-master/cget/cget/build/tmp-d2604558bab444c9b7165f9fa45fe4ed/xz-5.2.3':
configure: error: C compiler cannot create executables
See `config.log' for more details
-- Configuring done (2.9s)
-- Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CGET_CMAKE_DIR
    CGET_CMAKE_ORIGINAL_SOURCE_FILE

-- Build files have been written to: /Users/wellaglover/Documents/Codes RP/savvy-master/cget/cget/build/tmp-d2604558bab444c9b7165f9fa45fe4ed/build
[100%] Building xz ...
make[3]: *** No targets specified and no makefile found.  Stop.
gmake[2]: *** [CMakeFiles/xz.dir/build.make:71: CMakeFiles/xz] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/xz.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
Command failed: ['/usr/local/bin/cmake', '--build', '/Users/wellaglover/Documents/Codes RP/savvy-master/cget/cget/build/tmp-d2604558bab444c9b7165f9fa45fe4ed/build', '--config', 'Release', '--', '-j', '8']
Failed to build package statgen/savvy

**Comment by jonathonl** [comment]
Where are you downloading SAIGE from? The XZ package should be completely ignored with https://github.com/weizhouUMICH/SAIGE/blob/master/configure#L21.

**Comment by Welz2** [comment]
I'm downloading it through Rstudio and terminal 

I using the cget to download it with this  : cget install --prefix ./cget/ statgen/savvy

**Comment by jonathonl** [comment]
On recent version of MacOS, you need to run 
```
cget ignore --prefix ./cget/ xz 
```
before running
```
cget install --prefix ./cget/  statgen/savvy
```

**Comment by Welz2** [comment]
Thank you! I've run the above code but errors are still occurring

Wellas-MBP:savvy-master wellaglover$ cget ignore --prefix ./cget/ xz 
Ignore package xz
Wellas-MBP:savvy-master wellaglover$ cget install --prefix ./cget/  statgen/savvy
Downloading https://github.com/statgen/savvy/archive/HEAD.tar.gz
  [######################################################################]  100%
Extracting archive /Users/wellaglover/Documents/Codes RP/savvy-master/cget/cget/build/tmp-58b9b7148f2f4e9ca2a287beac2841c1/HEAD.tar.gz ...
Downloading https://github.com/jonathonl/shrinkwrap/archive/v1.2.0.tar.gz
  [######################################################################]  100%
Extracting archive /Users/wellaglover/Documents/Codes RP/savvy-master/cget/cget/build/tmp-50b0456bf4f64b3c8f3dda13c6725d23/v1.2.0.tar.gz ...
Downloading https://github.com/facebook/zstd/archive/v1.3.2.tar.gz
  [######################################################################]  100%
Extracting archive /Users/wellaglover/Documents/Codes RP/savvy-master/cget/cget/build/tmp-c4d84f398b0d4df9936cdfeddc19f643/v1.3.2.tar.gz ...
CMake Warning (dev) at CMakeLists.txt:10 (PROJECT):
  cmake_minimum_required() should be called prior to this top-level project()
  call.  Please see the cmake-commands(7) manual for usage documentation of
  both commands.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- The C compiler identification is AppleClang 14.0.3.14030022
-- The CXX compiler identification is AppleClang 14.0.3.14030022
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Deprecation Warning at CMakeLists.txt:11 (CMAKE_MINIMUM_REQUIRED):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


ENABLING STATIC
-- Performing Test CXX_FLAG_STD_CPLUSPLUS11
-- Performing Test CXX_FLAG_STD_CPLUSPLUS11 - Success
-- Performing Test C_FLAG_STD_C99
-- Performing Test C_FLAG_STD_C99 - Success
-- Performing Test C_FLAG_WALL
-- Performing Test C_FLAG_WALL - Success
-- Performing Test CXX_FLAG_WALL
-- Performing Test CXX_FLAG_WALL - Success
-- Performing Test C_FLAG_WEXTRA
-- Performing Test C_FLAG_WEXTRA - Success
-- Performing Test CXX_FLAG_WEXTRA
-- Performing Test CXX_FLAG_WEXTRA - Success
-- Performing Test C_FLAG_WUNDEF
-- Performing Test C_FLAG_WUNDEF - Success
-- Performing Test CXX_FLAG_WUNDEF
-- Performing Test CXX_FLAG_WUNDEF - Success
-- Performing Test C_FLAG_WSHADOW
-- Performing Test C_FLAG_WSHADOW - Success
-- Performing Test CXX_FLAG_WSHADOW
-- Performing Test CXX_FLAG_WSHADOW - Success
-- Performing Test C_FLAG_WCAST_ALIGN
-- Performing Test C_FLAG_WCAST_ALIGN - Success
-- Performing Test CXX_FLAG_WCAST_ALIGN
-- Performing Test CXX_FLAG_WCAST_ALIGN - Success
-- Performing Test C_FLAG_WCAST_QUAL
-- Performing Test C_FLAG_WCAST_QUAL - Success
-- Performing Test CXX_FLAG_WCAST_QUAL
-- Performing Test CXX_FLAG_WCAST_QUAL - Success
-- Performing Test C_FLAG_WSTRICT_PROTOTYPES
-- Performing Test C_FLAG_WSTRICT_PROTOTYPES - Success
-- ZSTD_LEGACY_SUPPORT not defined!
ZSTD VERSION 1.3.2
-- Configuring done (4.7s)
-- Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CGET_CMAKE_DIR
    CGET_CMAKE_ORIGINAL_SOURCE_FILE


-- Build files have been written to: /Users/wellaglover/Documents/Codes RP/savvy-master/cget/cget/build/tmp-c4d84f398b0d4df9936cdfeddc19f643/build
[  3%] Creating pkg-config file
[  7%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/common/fse_decompress.c.o
[ 11%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/common/threading.c.o
[ 15%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/common/entropy_common.c.o
[ 19%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/common/pool.c.o
[ 23%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/common/xxhash.c.o
[ 26%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/common/zstd_common.c.o
[ 30%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/common/error_private.c.o
CMake Error: File /Users/wellaglover/Documents/Codes/ RP/savvy-master/cget/cget/build/tmp-c4d84f398b0d4df9936cdfeddc19f643/zstd-1.3.2/lib/libzstd.pc.in does not exist.
CMake Error at /Users/wellaglover/Documents/Codes RP/savvy-master/cget/cget/build/tmp-c4d84f398b0d4df9936cdfeddc19f643/zstd-1.3.2/build/cmake/lib/pkgconfig.cmake:1 (CONFIGURE_FILE):
  CONFIGURE_FILE Problem configuring file


gmake[2]: *** [build/cmake/lib/CMakeFiles/libzstd.pc.dir/build.make:71: build/cmake/lib/CMakeFiles/libzstd.pc] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:154: build/cmake/lib/CMakeFiles/libzstd.pc.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
[ 34%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/compress/fse_compress.c.o
[ 38%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/compress/huf_compress.c.o
[ 42%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/compress/zstdmt_compress.c.o
[ 46%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/compress/zstd_compress.c.o
[ 50%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/compress/zstd_fast.c.o
[ 53%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/compress/zstd_double_fast.c.o
[ 57%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/compress/zstd_lazy.c.o
[ 61%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/compress/zstd_opt.c.o
[ 65%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/compress/zstd_ldm.c.o
[ 69%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/decompress/huf_decompress.c.o
[ 73%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/decompress/zstd_decompress.c.o
[ 76%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/dictBuilder/cover.c.o
[ 80%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/dictBuilder/divsufsort.c.o
[ 84%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/dictBuilder/zdict.c.o
[ 88%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/deprecated/zbuff_common.c.o
[ 92%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/deprecated/zbuff_compress.c.o
[ 96%] Building C object build/cmake/lib/CMakeFiles/libzstd_static.dir/__/__/__/lib/deprecated/zbuff_decompress.c.o
[100%] Linking C static library libzstd.a
[100%] Built target libzstd_static
gmake: *** [Makefile:136: all] Error 2
Command failed: ['/usr/local/bin/cmake', '--build', '/Users/wellaglover/Documents/Codes RP/savvy-master/cget/cget/build/tmp-c4d84f398b0d4df9936cdfeddc19f643/build', '--config', 'Release', '--', '-j', '8']
Failed to build package statgen/savvy

**Comment by jonathonl** [comment]
I've never seen zstd fail to build. Unfortunately, MacOS hasn't been very dependable in recent years and I have no control over third party dependencies failing to build on MacOS. You will have to install libzstd (maybe through homebrew or with cget using a newer version of zstd).   You would ignore zstd as well if installing it without cget. 

Ultimately, you will have to tell SAIGE how to find these dependencies. I believe it's looking for them in a specific place in the SAIGE configure script (which installs dependencies automatically). Installing dependencies in a different location is going to make things more difficult.

**Comment by Welz2** [comment]
Ok thank you. I'll try that advice and let you know of any updates!
Again thank you for responding :)!

**Comment by Welz2** [comment]
Hi just updating so I successfully able to download savvy in my terminal.
For MacOS Ventura -Mac M1
The way it worked:
 1-I downloaded the zip file from GitHub and then dragged the folder into my application folder.
2- I went into the directory of the savvy-master folder in my terminal
`cd /Applications/savvy-master `
3-  I ran the said method above about ignoring the package `cget ignore --prefix ./cget/ xz `
4-  Then I ran `cget install --prefix ./cget/  statgen/savvy`
5- It successfully downloaded savvy.

Now I think it could possibly where I had saved it before which was in Documents but when I moved the file to Applications it seemed to work.

Again thank you very much for helping me out!! :))
